IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <AuthenticationServices/AuthenticationServices.h>
+#import "ASAuthorizationSingleSignOnProvider+MSIDExtensions.h"
+#import "MSIDSSOExtensionInteractiveTokenRequest.h"
+#import "MSIDInteractiveTokenRequest+Internal.h"
+#import "MSIDJsonSerializer.h"
+#import "MSIDInteractiveRequestParameters.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDAuthority.h"
+#import "MSIDSSOExtensionTokenRequestDelegate.h"
+#import "MSIDBrokerOperationInteractiveTokenRequest+InteractiveParameters.h"
+#import "MSIDBrokerOperationInteractiveTokenRequest+SSORequest.h"
+
+@interface MSIDSSOExtensionInteractiveTokenRequest () <ASAuthorizationControllerPresentationContextProviding>
+
+@property (nonatomic) ASAuthorizationController *authorizationController;
+@property (nonatomic, copy) MSIDInteractiveRequestCompletionBlock requestCompletionBlock;
```

Comment:
wouldn't base class already store the completion block? if yes, should the property be shared and exposed in internal header?

Reply:
we don't store it in the base class.